internal/pkg/manager/nodestatus/nodestatus.go

```diff
@@ -0,0 +1,286 @@
+	if getProfErr != nil {
+		return reconcile.Result{}, getProfErr
+	}
+
+	lprof := logger.WithValues(
+		"Profile.Name", prof.GetName(),
+		"Profile.Namespace", prof.GetNamespace(),
+		"Profile.Kind", prof.GetObjectKind().GroupVersionKind(),
+	)
+
+	// get all the other statuses
+	// TODO: Delete status... it's just plain wrong
+	profName := instance.Labels[statusv1alpha1.StatusToProfLabel]
+	if profName == "" {
+		return reconcile.Result{}, errors.New("Unlabeled node status")
+	}
+
+	// TODO: Delete status... it's just plain wrong
+	if prof.GetName() != profName {
+		return reconcile.Result{}, errors.New("status doesn't match owner")
+	}
+
+	// Initialize status if it hasn't happened already
+	if prof.GetStatusBase().Status == "" {
+		lprof.Info("Initializing Profile status")
+
+		targetStatus := statusv1alpha1.ProfileStatePending
+		if instance.Status != "" {
+			targetStatus = instance.Status
+		}
+		return r.reconcileStatus(ctx, prof, targetStatus, lprof)
+	}
+
+	nodeStatusList, err := listStatusesForPolicy(ctx, r.client, profName)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "cannot list the node statuses")
+	}
+
+	// get the DS
+	spodDS, err := r.getDS(ctx, config.GetOperatorNamespace(), lprof)
+	if err != nil {
+		return reconcile.Result{}, errors.Wrap(err, "cannot get the DS")
+	}
+
+	if !daemonSetIsReady(spodDS) || daemonSetIsUpdating(spodDS) {
+		// If the DS is not ready or updating, don't bother updating the status
+		// TODO(jhrozek): is this needed? It sounds safer..
+		logger.Info("Not updating policy because the SPOd is not ready")
+		return reconcile.Result{RequeueAfter: dsWait}, nil
+	}
+
+	// make sure we have all the statuses already
+	hasStatuses := len(nodeStatusList.Items)
+	wantsStatuses := spodDS.Status.DesiredNumberScheduled
+	if wantsStatuses != int32(hasStatuses) {
+		logger.Info("Not updating policy: not all statuses are ready",
+			"has", hasStatuses, "wants", wantsStatuses)
+		// Don't reconcile again, let's just wait for another update
+		return reconcile.Result{}, nil
+	}
+
+	var lowestCommonState statusv1alpha1.ProfileState = statusv1alpha1.LowestState
+	for i := range nodeStatusList.Items {
+		lowestCommonState = statusv1alpha1.LowerOfTwoStates(lowestCommonState, nodeStatusList.Items[i].Status)
+	}
+	logger.Info("Setting the status to", "newState", lowestCommonState)
+
+	return r.reconcileStatus(ctx, prof, lowestCommonState, lprof)
+}
+
+func (r *StatusReconciler) getDS(ctx context.Context, namespace string, l logr.Logger) (*appsv1.DaemonSet, error) {
+	dsSelect := labels.NewSelector()
+	spodDSFilter, err := labels.NewRequirement("spod", selection.Exists, []string{})
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create DS list label")
+	}
+	dsSelect.Add(*spodDSFilter)
+	dsListOpts := client.ListOptions{
+		LabelSelector: dsSelect,
+		Namespace:     namespace,
+	}
+
+	spodDSList := appsv1.DaemonSetList{}
+	if err := r.client.List(ctx, &spodDSList, &dsListOpts); err != nil {
```

Comment:
Don't we know its name? Why do a `List` instead of a `Get`?

Reply:
Mind if I update this in anther PR?

Reply:
Sure :+1: